recipes/mongo-cxx-driver/all/conanfile.py

```diff
@@ -0,0 +1,112 @@
+    exports_sources = ["CMakeLists.txt", "patches/**", "Find*.cmake"]
+    generators = ("cmake", "cmake_find_package")
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "polyfill": ["std", "boost", "mnmlstc", "experimental"]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "polyfill": "boost"
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if self.settings.compiler == "Visual Studio" and self.options.polyfill != "boost":
+            raise ConanInvalidConfiguration("For MSVC, best to use the boost polyfill")
+
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, "17" if self.options.polyfill == "std" else "11")
+
+    def requirements(self):
+        self.requires("mongo-c-driver/1.17.2")
+        if self.options.polyfill == "boost":
+            self.requires("boost/1.74.0")
+
+        # Cannot model mnmlstc (not packaged, is pulled dynamically) or
+        # std::experimental (how to check availability in stdlib?) polyfill
+        # dependencies
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-r" + self.version, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        self._cmake.definitions["BSONCXX_POLY_USE_MNMLSTC"] = self.options.polyfill == "mnmlstc"
+        self._cmake.definitions["BSONCXX_POLY_USE_STD_EXPERIMENTAL"] = self.options.polyfill == "experimental"
+        self._cmake.definitions["BSONCXX_POLY_USE_BOOST"] = self.options.polyfill == "boost"
+        self._cmake.definitions["BUILD_VERSION"] = self.version
+        self._cmake.definitions["BSONCXX_LINK_WITH_STATIC_MONGOC"] = not self.options["mongo-c-driver"].shared
+        self._cmake.definitions["MONGOCXX_LINK_WITH_STATIC_MONGOC"] = not self.options["mongo-c-driver"].shared
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+    def build(self):
+        if self.settings.compiler == "Visual Studio":
+            cmake_file = os.path.join(self._source_subfolder, "../CMakeLists.txt")
+            tools.replace_in_file(
+                cmake_file,
+                'add_subdirectory("source_subfolder")',
+                'add_subdirectory("source_subfolder")\nadd_definitions(-D_ENABLE_EXTENDED_ALIGNED_STORAGE)'
```

Comment:
Where does this define come from?
I just grepped for `ENABLE_EXTENDED_ALIGNED_STORAGE` in the sources of `mongo-cxx-driver` and did not get any match.

Reply:
It' something VS specific.

https://developercommunity.visualstudio.com/content/problem/425273/enabling-enable-extended-aligned-storage-in-a-mixe.html

I took it from the bincrafters recipe.

I will try to remove it, but I'm no Windows expert at all.

Reply:
I decided to make the recipe work and pass the CI first, and then try to remove the corresponding lines. It will be easier for me to see if this ENABLE_EXTENDED_ALIGNED_STORAGE necessary.

Reply:
Deleted these flags